src/main/kotlin/io/thecontext/ci/context/OutputContext.kt

```diff
@@ -11,13 +11,15 @@ interface OutputContext : Context {
 
     class Impl(context: Context) : OutputContext, Context by context {
 
-        private val markdownRenderer by lazy { HtmlRenderer.Impl() }
-        private val mustacheRenderer by lazy { TemplateRenderer.Impl() }
+        private val htmlRenderer by lazy { HtmlRenderer.Impl() }
```

Comment:
NIT: For all the `val` in this class: you use `by lazy { ... } `  to create the instance BUT you return the implementation hence all fields of  `OutputContext.Impl` are of the implementation type instead of the interface. example:
```kotlin
private val htmlRenderer by lazy { HtmlRenderer.Impl() }   // it's actually of type  private val htmlRenderer : HtmlRenderer.Impl
```

I would suggest to add the interface type definition to all fields of this class:

```kotlin
private val htmlRenderer : HtmlRenderer by lazy { HtmlRenderer.Impl() }
private val templateRenderer  : TemplateRenderer  by lazy { TemplateRenderer.Impl() }
private val textWriter : TextWriter by lazy { TextWriter.Impl() }
...
```

Reply:
Same for `InputContext.Impl`

Reply:
This is a totally valid point, but I see only a single issue this approach solves. `*Context` users have to deal with the interface since it is declared in the `*Context` signature. Implementations declared in the `*Context` have interfaces as constructor arguments as well. So, the only issue I see is some sneaking component requiring an implementation instead of an interface as an argument. Do you mean this one or do you have anything else in mind?

Going to apply it anyway just to be sure.